services/embeddings/generate_embeddings.py

```diff
@@ -0,0 +1,182 @@
+        return EmbeddingClass()
+    except KeyError:
+        raise ValueError(f"Unsupported embedding type: {embedding_type}")
+
+def _get_vectorstore_config(vectorstore_type='zilliz', connection_args=None, **kwargs):
+    """
+    Internal helper to get vectorstore class and configuration.
+    
+    Args:
+        vectorstore_type: Name of vectorstore class (lowercase)
+        connection_args: Dictionary of connection arguments for specific vectorstores (e.g. Zilliz)
+        **kwargs: Additional parameters for vectorstore initialisation
+    """
+    vectorstore_classes = {
+        'zilliz': Zilliz
+        # Test other vectorstore classes from LangChain and add here
+    }
+    
+    try:
+        VectorStoreClass = vectorstore_classes[vectorstore_type.lower()]
+        
+        # Build kwargs dictionary
+        init_kwargs = {
+            # 'embedding_function': embedding,
+            **kwargs
+        }
+        
+        # Add connection args if provided
+        if connection_args:
+            init_kwargs['connection_args'] = connection_args # keeping in here instead of kwargs to flag that it might be needed for some dbs
+            
+        return VectorStoreClass, init_kwargs
+        
+    except KeyError:
+        raise ValueError(f"Unsupported vectorstore type: {vectorstore_type}")
+
+def create_vectorstore(docs, collection_name='LangChainCollection', vectorstore_type='zilliz', embedding="openai",
+                        connection_args=None, auto_id=True, drop_old=True, **kwargs):
+    """
+    Create a new vectorstore from documents with flexible vectorstore selection.
+    """
+    VectorStoreClass, init_kwargs = _get_vectorstore_config(
+        vectorstore_type=vectorstore_type,
+        # embedding=embedding,
+        connection_args=connection_args,
+        collection_name=collection_name,
+        **kwargs
+    )
+    
+    return VectorStoreClass.from_documents(documents=docs, embedding=_get_embedding_class(embedding), auto_id=auto_id, drop_old=drop_old, **init_kwargs)
+
+def get_existing_vectorstore(collection_name='LangChainCollection', vectorstore_type='zilliz', 
+                           embedding="openai", connection_args=None, auto_id=True, **kwargs):
+    """
+    Get existing vectorstore collection with flexible vectorstore selection.
+
+    Args:
+    ...
+
+    """
+    VectorStoreClass, init_kwargs = _get_vectorstore_config(
+        vectorstore_type=vectorstore_type,
+        # embedding=embedding,
+        connection_args=connection_args,
+        collection_name=collection_name,
+        **kwargs
+    )
+    
+    return VectorStoreClass(embedding_function=_get_embedding_class(embedding), auto_id=auto_id, **init_kwargs)
+
+
+def get_similar_texts(input_text, vectorstore, search_type="similarity", search_kwargs={"k": 2}):
+    """
+    Retrieve similar texts from a vectorstore based on input text.
+    
+    Args:
+        input_text: Text string to use for similarity search
```

Comment:
Ah I'm interested in this - so we don't need to manually generate embeddings for our search strings? Langchain does this for us based on the built-in embedding function?

This is an area I've been curious about.

Reply:
No we don't, although we could if we wanted to. It'll just use the embedding type associated with the vector store.